File: LeetCode/python/MergeTwoSortedList.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def mergeTwoLists(self, list1: Optional[ListNode], list2: Optional[ListNode]) -> Optional[ListNode]:
        head = ListNode()
        if list1!=None and list2!=None:
            if list1.val <= list2.val :
                head.val = list1.val
                head.next = None
                list1 = list1.next
            else:
                head.val = list2.val
                head.next = None
                list2 = list2.next
        it = head
        while list1!=None and list2!=None:
            if list1.val <= list2.val :
                it.next.val = list1.val
                list1 = list1.next
                it = it.next
            else:
                it.next.val = list2.val
                list1 = list1.next
                it = it.next

        if list1!=None:
            while list1!=None:
                it.next.val = list1.val
                list1 = list1.next
                it = it.next
        if list1!=None:
            while list2!=None:
                it.next.val = list2.val
                list1 = list2.next
                it = it.next

        return head

File: LeetCode/JavaScript/MergeTwoSortedList.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} list1
 * @param {ListNode} list2
 * @return {ListNode}
 */
var mergeTwoLists = function(list1, list2) {
    
    if(!list1 || !list2 )
        return !list1? list2 : list1; 
    
    let head = new ListNode();
    if(list1.val<=list2.val){
        head = list1;
        list1 = list1.next;
    }
    else {
        head = list2;
        list2 = list2.next;
    }
    let temp = head;
    
    while(list1 && list2){
        if(list1.val<= list2.val){
            temp.next = list1;
            temp = temp.next;
            list1 = list1.next;
        }
        else{
            temp.next = list2;
            temp = temp.next;
            list2 = list2.next;
        }
    }

    temp.next = list1? list1 : list2;

    return head
};